### DISCUSSION TITLE
Authorization header not being passed to serverless function endpoint (Astro project)

### DISCUSSION THREAD

**Discussion by JayBox325** [vercel/community#6032] [Help]
### Summary

I have an Astro project where a user submits a form to sign into their Firebase auth account, this submit function then goes to `/api/auth/signin` to then run the Firebase signin functionality by passing the Firebase user idToken to the endpoint using the Authorization token.

This process works perfectly locally, however, when I push it up to Vercel, the feedback in the console (from the logs I've added), the response is that `request.headers` are `null`.

And I'm not sure why there are no request headers being passed to the API endpoint and I'm struggling to find any resources on this.

Here's the function where the request is being made:
```
try {
        const res = await fetch("/api/auth/signin", {
            method: "GET",
            headers: {
                Authorization: `Bearer ${idToken}`,
            },
        });

        const response = await res.json()
        console.log('response', response)

        // Redirect to the account page
        // window.location.href = "/account";

        // Reset the form
        formErrorEl?.classList.add("hidden")
        formErrorEl.textContent = ""
    } catch (error) {
        console.log("❌ Sign in error", error)
        formErrorEl?.classList.remove("hidden")
        formErrorEl.textContent = error.code;
    }
```

And here is an excerpt from the API endpoint:
```
export const GET = async ({ request, cookies }) => {
    const auth = getAuth(app)

    /* Get token from request headers */
    const idToken = request.headers.get("Authorization")?.split("Bearer ")[1];

    console.log('id Token', idToken)

    if (!idToken) { 
        return new Response(
            JSON.stringify({ message: "❌ No token found", code: "missing token" }),
            { status: 401, headers: { 'Content-Type': 'application/json' }
        });
    }
}
```

Any help would be greatly appreciated!

### Steps to Reproduce

With an astro project, set up an API endpoint using the Astro docs then deploy it to Vercel.

**Comment by igorlanko** [comment]
How did you manage to figure this out? I'm trying to understand why Vercel strips authorization headers selectively. For example in my case it passes it when chrome extension is making a request, but not when the desktop app does it.

[a comment by an automated account is not shown]
